src/Collection/Collection.model.ts

```diff
@@ -101,4 +101,35 @@ export class Collection extends Model<CollectionAttributes> {
 
     return counts.length > 0 && counts[0].count <= 0
   }
+
+  static async upsertWithItemCount(
+    collection: CollectionAttributes
+  ): Promise<CollectionWithItemCount> {
+    const { id, ...attributes } = collection
+    const columnNames = ['id', ...Object.keys(attributes)]
+    const columnValues = [id, ...Object.values(attributes)]
+
+    const insertColumns = columnNames.map((name) => `"${name}"`)
+    const insertValues = columnNames.map((_, index) => `$${index + 1}`)
+
+    const updateFields = columnNames
```

Comment:
Can we use the `toAssignmentFields` method as we're doing [here](https://github.com/decentraland/builder-server/blob/54ba34e78b691a5ca789d0c5b8c7cdcb53430cf6/src/Curation/Curation.service.ts#L81)?